compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/QueryTypeChecker.java

```diff
@@ -529,26 +532,38 @@ private BType getResolvedType(BType initType, BType expType, boolean isReadonly,
         return initType;
     }
 
-    private BType getNonContextualQueryType(BType staticType, BType basicType) {
-        BType resultType;
-        switch (Types.getReferredType(basicType).tag) {
+    private BType getNonContextualQueryType(BType staticType, BType basicType, Location pos) {
+        basicType = Types.getReferredType(basicType);
```

Comment:
If we have a type reference as the basic type, the error will have the referred type if we change it here